src/Controller/CustomerController.php

```diff
@@ -83,6 +83,28 @@ public function create(
         
     }
 
+    /**
+     * @Route("/api/customers/{id}", name="delete_customer", methods={"DELETE"})
+     */
+    public function delete(?Customer $customer, EntityManagerInterface $manager, SupplierRepository $supplierRepository)
+    {
+        /** Replace it when auth wil be implemented */
+        $supplier = $this->getSupplier($supplierRepository);
+
+        if(!$customer){
+            return $this->json(['message' => 'Resource introuvable'], JsonResponse::HTTP_NOT_FOUND);
+        }
+
+        if($customer->getSupplier()->getId() !== $supplier->getId()){
+            return $this->json(['message' => 'Vous n\'êtes pas authorisé à supprimer ce client!'], JsonResponse::HTTP_FORBIDDEN);
+        }
+
+        $manager->remove($customer);
+        $manager->flush();
+
+        return $this->json(['message' => 'Le client à été supprimé'], JsonResponse::HTTP_NO_CONTENT);
```

Comment:
no content cannot have body. it should be an empty response

Reply:
Fixed in authentication branch